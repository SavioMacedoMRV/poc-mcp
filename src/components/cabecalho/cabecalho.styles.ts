import styled, { css } from 'styled-components'
import {cores} from 'resources/cores'
import {media} from 'resources/media'
import {margens} from 'resources/margens'
import {Texto} from 'components/texto'
import {BotaoIcone} from 'components/botaoIcone'
import {BotaoNavegacao} from 'components/botaoNavegacao'
import {Input} from 'components/input'
import {Botao} from 'components/botao'
import ChevronSVG from 'icons/iChevron.svg'
import FecharSVG from 'icons/iFechar.svg'
import FiltroSVG from 'icons/iFiltro.svg'
import LupaSVG from 'icons/iLupa.svg'
import BlocoSVG from 'icons/iBloco.svg'
import ArrowBackIosSVG from 'icons/iArrowBackIos.svg'

export const BotaoSair = styled(Botao).attrs(props => ({
  texto: 'Sair',
  corTexto: cores.neutralXXLight,
  textoProps: {
    tamanho: 12,
    estilo: 'semibold',
    alturalinha: 20
  },
  iconeProps: {
    altura: 24,
    largura: 24,
    cor: cores.neutralXXLight
  },
  espaco: 4,
  icone: FecharSVG,
  cor: cores.supportFeedbackError,
  ...props,
}))`
  width: 112px;
  border-radius: 8px;
  padding: 4px 8px;
  justify-content: center;
  height: fit-content;

  @media ${media.tablet} {
    span {
      display: none;
    }
  }
`

export const BotaoVoltar = styled(BotaoIcone).attrs({
  cor: cores.neutralXDark,
  altura: 24,
  largura: 24,
})`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 8px;
  border-radius: 4px;
  
  &:hover {
    background-color: ${cores.background01};
  }
`

export const Container = styled.header<{ $modoCadastro?: boolean }>`
  position: sticky;
  z-index: 1;
  top: 0px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 56px;
  background: ${cores.neutralXXLight};
  padding: 8px 64px;
  gap: 237px;
  border-bottom: 1px solid ${cores.background01};

  ${({ $modoCadastro }) => $modoCadastro && css`
    gap: 0;

    ${HeaderLeft} {
      ${BotaoVoltar} {
        display: flex;
        align-items: center;
        gap: 8px;
        padding: 4px;
        border-radius: 4px;
        height: 32px;
        
        &:hover {
          background-color: ${cores.background01};
        }
      }

      ${TextoVoltar} {
        font-size: 12px;
        line-height: 20px;
      }
    }
  `}

  @media ${media.tablet} {
    padding: 8px 16px;
    gap: 16px;
  }
`

export const HeaderCenter = styled.div`
  display: flex;
  align-items: center;
  gap: 24px;
  width: 608px;
  justify-content: center;

  @media ${media.tablet} {
    width: auto;
    gap: 16px;
  }
`

export const HeaderLeft = styled.div`
  display: flex;
  align-items: center;
  gap: 8px;
`

export const HeaderRight = styled.div`
  display: flex;
  justify-content: flex-end;
  align-items: center;
`

export const IconeFiltro = styled(FiltroSVG)`
  path {
    fill: ${cores.neutralXDark};
  }
`

export const IconeLocal = styled(BlocoSVG)`
  path {
    fill: ${cores.neutralXDark};
  }
`

export const IconePesquisa = styled(LupaSVG)`
  path {
    fill: ${cores.neutralXDark};
  }
`

export const IconeVoltar = styled(ArrowBackIosSVG)`
  path {
    fill: ${cores.neutralXDark};
  }
`

export const InputPesquisar = styled(Input).attrs({
  tipo: 'pesquisa',
  placeholder: 'Pesquisar',
  textoProps: {
    tamanho: 12,
    cor: cores.neutralXDark,
    estilo: 'regular'
  },
  icone: {
    largura: 24,
    altura: 24,
    icone: IconePesquisa
  },
  altura: 32,
  corBorda: cores.background01,
  corBordaFocada: cores.background01,
  iconeEsquerda: true,
})`
  width: 480px;
  border-width: 2px;
  border-radius: 8px;
  background: ${cores.neutralXXLight};
  height: 32px;
  padding: 0;
  padding-left: 10px;
  display: flex;
  align-items: center;
  
  div {
    align-items: center;
    gap: 8px;
    height: 100%;
    padding: 4px 8px;
    flex: 1;
  }

  input {
    font-family: 'AvertaStd-Regular';
    font-size: 12px;
    line-height: 1.235em;
    height: 100%;
    padding: 0;
  }

  @media ${media.tablet} {
    width: 100%;
  }
`

export const TextoVoltar = styled(Texto).attrs({
  tamanho: 12,
  alturalinha: 20,
  estilo: 'semibold',
  cor: cores.neutralXDark,
})`
  font-family: AvertaStd-Semibold;
  font-style: normal;
  font-weight: 400;
`

export const TextoObra = styled(TextoVoltar)``
export const TextoProjeto = styled(TextoVoltar)``

export const BotaoFiltro = styled(BotaoIcone).attrs({
  altura: 24,
  largura: 24,
  cor: cores.neutralXDark,
})`
  padding: 4px;
  width: 32px;
  height: 32px;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 4px;
  border: none;
  background: transparent;

  &:hover {
    background-color: ${cores.background01};
  }
`

export const BotaoHubs = styled(BotaoNavegacao)`
  && {
    width: 160px;
    height: 32px;
    padding: 4px 8px;
    border-radius: 8px;
    border: 2px solid ${cores.background01};
    background: ${cores.neutralXXLight};
    gap: 8px;

    span {
      font-family: 'AvertaStd-Regular';
      font-size: 12px;
      line-height: 1.235em;
      color: ${cores.neutralXDark};
    }

    &:hover {
      background-color: ${cores.background01};
    }
  }
`
