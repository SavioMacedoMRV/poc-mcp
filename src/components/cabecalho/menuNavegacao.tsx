import React from 'react'
import { useNavigate } from 'react-router-dom'
import { Texto } from 'components/texto'
import { rotas } from 'resources/rotas'
import styled from 'styled-components'
import BlocoSVG from 'icons/iBloco.svg'
import iSeta from 'icons/iSeta.svg'
import { Icone } from 'components/icone'
import { cores } from 'resources/cores'

const MenuItem = styled.button`
  display: flex;
  align-items: center;
  gap: 12px;
  background: none;
  border: none;
  cursor: pointer;
  padding: 8px 12px;
  width: 100%;
  text-align: left;
  transition: background-color 0.2s;
  border-radius: 8px;
  min-height: 40px;

  &:hover {
    background-color: ${cores.background01};
  }
`

const Divider = styled.div`
  height: 1px;
  background: ${cores.background01};
  width: 100%;
  margin: 4px 0;
`

const StyledIcone = styled(Icone)`
  path {
    fill: ${cores.neutralXDark};
  }
  flex-shrink: 0;
`

export const MenuNavegacao = () => {
  const navigate = useNavigate()

  const menuItems = [
    { label: 'Hubs', icon: BlocoSVG, route: rotas.HubFormas },
    { label: 'Obras', icon: iSeta, route: rotas.OutraPagina },
    { label: 'Fornecedores', icon: BlocoSVG, route: rotas.Home },
    { label: 'Pe√ßas', icon: BlocoSVG, route: rotas.Home }
  ]

  return (
    <>
      {menuItems.map((item, index) => (
        <React.Fragment key={item.label}>
          <MenuItem onClick={() => navigate(item.route)}>
            <StyledIcone icone={item.icon} altura={20} largura={20} />
            <Texto 
              tamanho={12} 
              estilo="semibold" 
              cor={cores.neutralXDark}
              alturalinha={20}
            >
              {item.label}
            </Texto>
          </MenuItem>
          {index < menuItems.length - 1 && <Divider />}
        </React.Fragment>
      ))}
    </>
  )
}