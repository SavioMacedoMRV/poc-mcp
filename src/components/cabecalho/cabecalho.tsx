import React, {ReactElement, useCallback, useState} from 'react'
import * as S from './cabecalho.styles'
import {Evento} from 'types/enums'
import {Empreendimento} from 'types/interfaces'
import {BotaoIcone} from 'components/botaoIcone'
import {RegistrarEvento} from 'infrastructure/logs'
import {useNavigate} from 'react-router-dom'
import {rotas} from 'resources/rotas'
import {cores} from 'resources/cores'
import {BotaoProps} from 'components/botao/botao.styles'
import {ModalConfirmacao} from 'components/modalConfirmacao'
import {MenuNavegacao} from './menuNavegacao'
import {Texto} from 'components/texto'
import ArrowBackIosSVG from '../../assets/icons/iArrowBackIos.svg'
import FecharSmallSVG from '../../assets/icons/iFecharSmall.svg'

export interface CabecalhoProps {
  obra?: Empreendimento | null
  pesquisa?: {
    autoPesquisar?: boolean
    aoPesquisar: (texto?: string | number) => void
  }
  filtroPepSuperior?: ReactElement
  filtroAtividade?: ReactElement
  filtroLocal?: ReactElement
  cabecalhoDireita?: ReactElement
  aoClicarVoltar?: () => void
  botaoSair?:
    | (BotaoProps & {confirmacao?: {titulo: string; mensagem: string}})
    | boolean
  modoCadastro?: boolean
}

export const Cabecalho = ({
  obra,
  cabecalhoDireita,
  pesquisa,
  filtroPepSuperior,
  filtroAtividade,
  filtroLocal,
  aoClicarVoltar,
  botaoSair,
  modoCadastro,
}: Readonly<CabecalhoProps>) => {
  const navigate = useNavigate()
  const projetoSelecionado = obterProjetoSelecionado(obra)
  const [exibirModalConfirmar, setExibirModalConfirmar] = useState(false)
  const [menuAberto, setMenuAberto] = useState(false)

  const aoClicarVoltarHandler = useCallback(() => {
    RegistrarEvento({evento: Evento.CabecalhoClickVoltar})
    if (aoClicarVoltar) {
      aoClicarVoltar()
      return
    }
    navigate(-1)
  }, [aoClicarVoltar, navigate])

  const aoFecharModalConfirmacao = () => setExibirModalConfirmar(false)

  const aoConfirmarSair = useCallback(() => {
    aoFecharModalConfirmacao()
    if (typeof botaoSair === 'object' && botaoSair.aoClicar) {
      botaoSair.aoClicar()
      return
    }
    navigate(rotas.Home)
  }, [botaoSair, navigate])

  const aoClicarSairHandler = useCallback(() => {
    RegistrarEvento({evento: Evento.CabecalhoClickSair})
    if (typeof botaoSair === 'object' && botaoSair.confirmacao) {
      setExibirModalConfirmar(true)
      return
    }
    aoConfirmarSair()
  }, [aoConfirmarSair, botaoSair])

  if (modoCadastro) {
    return (
      <S.Container data-testid="cabecalho" $modoCadastro>
        <S.HeaderLeft>
          <S.BotaoVoltar
            data-testid="btn-cabecalho-voltar"
            icone={S.IconeVoltar}
            aoClicar={aoClicarVoltarHandler}
          />
          <S.TextoVoltar>Voltar</S.TextoVoltar>
        </S.HeaderLeft>
        <S.HeaderRight>
          <S.BotaoSair
            data-testid="btn-cabecalho-sair"
            texto="Cancelar"
            icone={FecharSmallSVG}
            aoClicar={aoClicarVoltarHandler}
            cor={cores.complementary09}
            corTexto={cores.neutralXXLight}
            textoProps={{
              tamanho: 12,
              estilo: 'semibold',
              alturalinha: 20
            }}
            iconeProps={{
              altura: 24,
              largura: 24,
              cor: cores.neutralXXLight
            }}
          />
        </S.HeaderRight>
      </S.Container>
    )
  }

  return (
    <S.Container data-testid="cabecalho">
      <S.HeaderLeft>
        <S.BotaoVoltar
          data-testid="btn-cabecalho-voltar"
          icone={S.IconeVoltar}
          aoClicar={aoClicarVoltarHandler}
        />
        <S.TextoVoltar>
          {!projetoSelecionado ? 'Voltar' : `${projetoSelecionado.projeto} /`}
        </S.TextoVoltar>
        {projetoSelecionado && (
          <S.TextoVoltar>{projetoSelecionado.obra}</S.TextoVoltar>
        )}
      </S.HeaderLeft>
      <S.HeaderCenter>
        {pesquisa && (
          <S.InputPesquisar
            data-testid={'pesquisa-cabecalho'}
            aoDigitar={pesquisa.autoPesquisar ? pesquisa.aoPesquisar : undefined}
            aoSubmeter={pesquisa.aoPesquisar}
          />
        )}
        <S.BotaoHubs
          data-testid={'navegacao-hubs-cabecalho'}
          onSelect={() => {}}
        />
        {filtroAtividade && (
          <S.BotaoFiltro
            data-testid={'filtro-atividade-cabecalho'}
            icone={S.IconeFiltro}
            aoClicar={() => {}}
          />
        )}
        {filtroLocal && (
          <S.BotaoFiltro
            data-testid={'filtro-local-cabecalho'}
            icone={S.IconeLocal}
            aoClicar={() => {}}
          />
        )}
      </S.HeaderCenter>
      <S.HeaderRight>
        {cabecalhoDireita ?? 
          (botaoSair && (
            <S.BotaoSair
              data-testid="btn-cabecalho-sair"
              {...(typeof botaoSair === 'object' ? botaoSair : {})}
              aoClicar={aoClicarSairHandler}
            />
          ))}
        {typeof botaoSair === 'object' && botaoSair.confirmacao && (
          <ModalConfirmacao
            exibir={exibirModalConfirmar}
            titulo={botaoSair.confirmacao.titulo}
            mensagem={botaoSair.confirmacao.mensagem}
            aoConfirmar={aoConfirmarSair}
            aoCancelar={aoFecharModalConfirmacao}
          />
        )}
      </S.HeaderRight>
    </S.Container>
  )
}

export const obterProjetoSelecionado = (obra?: Empreendimento | null) => {
  if (obra) {
    return {obra: obra.nome, projeto: 'Projeto Executivo'}
  }
  return undefined
}
