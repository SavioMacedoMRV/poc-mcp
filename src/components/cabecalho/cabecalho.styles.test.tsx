import {render} from '@testing-library/react'
import * as S from './cabecalho.styles'

describe('Cabecalho.styles', () => {
  test('BotaoSair DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.BotaoSair aoClicar={jest.fn} />)

    expect(container).toMatchSnapshot()
  })

  test('BotaoVoltar DEVE ser igual ao snapshot', () => {
    const {container} = render(
      <S.BotaoVoltar icone={S.IconeVoltar} aoClicar={jest.fn} />,
    )

    expect(container).toMatchSnapshot()
  })

  test('Container DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.Container />)
    expect(container).toMatchSnapshot()
  })

  test('HeaderCenter DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.HeaderCenter />)
    expect(container).toMatchSnapshot()
  })

  test('HeaderLeft DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.HeaderLeft />)
    expect(container).toMatchSnapshot()
  })

  test('HeaderRight DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.HeaderRight />)
    expect(container).toMatchSnapshot()
  })

  test('IconeFiltro DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.IconeFiltro />)
    expect(container).toMatchSnapshot()
  })

  test('IconeLocal DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.IconeLocal />)
    expect(container).toMatchSnapshot()
  })

  test('IconeVoltar DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.IconeVoltar />)
    expect(container).toMatchSnapshot()
  })

  test('InputPesquisar DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.InputPesquisar />)
    expect(container).toMatchSnapshot()
  })

  test('TextoObra DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.TextoObra />)
    expect(container).toMatchSnapshot()
  })

  test('TextoProjeto DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.TextoProjeto />)
    expect(container).toMatchSnapshot()
  })
})
