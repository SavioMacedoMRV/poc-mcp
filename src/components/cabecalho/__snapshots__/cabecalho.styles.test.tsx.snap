// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cabecalho.styles BotaoSair DEVE ser igual ao snapshot 1`] = `
.c2 {
  font-weight: 400;
  font-size: 14px;
  line-height: 24px;
  font-family: AvertaStd-Semibold;
  color: #F4715D;
  text-align: start;
  cursor: pointer;
  text-transform: none;
}

.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  border-radius: 8px;
  box-sizing: border-box;
  padding: 4px 16px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  gap: 16px;
  width: auto;
  height: auto;
  cursor: pointer;
  opacity: 1;
  background: #F7F5F2;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c0:disabled {
  background: #D9DDDB;
}

@media (max-width: 768px) {
  .c1 span {
    display: none;
  }
}

<div>
  <button
    class="c0 c1"
    cor="#F7F5F2"
  >
    <span
      alturalinha="24"
      class="c2 "
      cor="#F4715D"
      cursor="pointer"
      data-testid="texto-botao"
      estilo="semibold"
      tamanho="14"
    >
      Sair
    </span>
    <svg
      data-file-name="SvgIFechar"
      fill="#F4715D"
      height="24"
      style="min-width: 24px; min-height: 24px; max-width: 24px; max-height: 24px; transform: rotate(0deg);"
      width="24"
    />
  </button>
</div>
`;

exports[`Cabecalho.styles BotaoVoltar DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  border-radius: 8px;
  border: none;
  align-items: center;
  justify-content: center;
  background: transparent;
  padding: 8px;
  opacity: 1;
  cursor: pointer;
}

.c0:hover {
  opacity: 0.7;
}

.c0:active {
  opacity: 1;
}

.c0:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c1 {
  margin-right: 4px;
}

@media (max-width: 768px) {
  .c1 {
    margin-left: 0px;
  }
}

<div>
  <button
    class="c0 c1"
    cor="#5D615F"
  >
    <svg
      class=""
      data-file-name="SvgIChevron"
      data-testid="icone"
      fill="#5D615F"
      height="40"
      style="transform: rotate(90deg);"
      width="40"
    />
  </button>
</div>
`;

exports[`Cabecalho.styles Container DEVE ser igual ao snapshot 1`] = `
.c0 {
  position: sticky;
  z-index: 1;
  top: 0px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 64px;
  background: #FFFFFF;
  padding: 0 64px;
}

@media (max-width: 768px) {
  .c0 {
    padding: 0 24px;
  }
}

<div>
  <header
    class="c0"
  />
</div>
`;

exports[`Cabecalho.styles HeaderCenter DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  flex: 3;
  align-items: center;
  justify-content: center;
  gap: 16px;
  margin-left: 24px;
}

@media (max-width: 768px) {
  .c0 {
    margin-left: 0;
  }
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Cabecalho.styles HeaderLeft DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  flex: 2;
  align-items: center;
  margin-left: 24px;
}

@media (max-width: 768px) {
  .c0 {
    margin-left: 0;
  }
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Cabecalho.styles HeaderRight DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  flex: 2;
  justify-content: flex-end;
  align-items: center;
  margin-right: 24px;
}

@media (max-width: 768px) {
  .c0 {
    margin-right: 0;
  }
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Cabecalho.styles IconeFiltro DEVE ser igual ao snapshot 1`] = `
<div>
  <svg
    class=""
    data-file-name="SvgIFiltro"
  />
</div>
`;

exports[`Cabecalho.styles IconeLocal DEVE ser igual ao snapshot 1`] = `
<div>
  <svg
    class=""
    data-file-name="SvgIBloco"
  />
</div>
`;

exports[`Cabecalho.styles IconeVoltar DEVE ser igual ao snapshot 1`] = `
<div>
  <svg
    class=""
    data-file-name="SvgIChevron"
  />
</div>
`;

exports[`Cabecalho.styles InputPesquisar DEVE ser igual ao snapshot 1`] = `
.c3 {
  display: flex;
  border-radius: 8px;
  border: none;
  align-items: center;
  justify-content: center;
  background: transparent;
  padding: 0px;
  opacity: 1;
  cursor: pointer;
}

.c3:hover {
  opacity: 0.7;
}

.c3:active {
  opacity: 1;
}

.c3:focus-visible {
  outline: unset;
  border: 2px solid #999E9C;
}

.c0 {
  display: flex;
  padding: 8px;
  gap: 8px;
  flex-direction: row-reverse;
  justify-content: flex-end;
  height: 24px;
  width: auto;
  background: #FFFFFF;
  border-radius: 8px;
  border: 2px solid #F2F2F2;
}

.c0:focus-within {
  border: 2px solid #38C78B;
  box-shadow: 0px 2px 8px 0px rgba(67, 70, 69, 0.3);
}

.c2 {
  border: none;
  outline: none;
  background: transparent;
  width: 100%;
  font-size: 12px;
  line-height: normal;
  font-family: AvertaStd-Semibold;
  color: #242625;
  text-align: start;
  cursor: text;
  text-transform: none;
}

.c2:focus::placeholder {
  color: transparent;
}

.c2:disabled {
  color: #999E9C;
  cursor: default;
}

.c2::-webkit-search-cancel-button {
  -webkit-appearance: none;
  background: #999E9C;
  height: 10px;
  width: 10px;
  border-radius: 5px;
}

.c2::-webkit-inner-spin-button {
  -webkit-appearance: none;
}

.c2::-webkit-outer-spin-button {
  -webkit-appearance: none;
}

.c4 {
  align-items: baseline;
}

.c1 {
  flex: 1;
  padding: 4px 8px;
  max-width: 720px;
}

<div>
  <div
    altura="24"
    class="c0 c1"
    data-testid="input"
  >
    <input
      class="c2"
      data-testid="input-input"
      placeholder="Pesquisar"
      tamanho="12"
      value=""
    />
    <button
      class="c3 c4"
      cor="#434645"
      data-testid="btn-input-pesquisar"
    >
      <svg
        data-file-name="SvgIPesquisar"
        data-testid="icone"
        fill="#434645"
        height="24"
        style="transform: rotate(0deg);"
        width="24"
      />
    </button>
  </div>
</div>
`;

exports[`Cabecalho.styles TextoObra DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  font-family: AvertaStd-Semibold;
  color: #793F98;
  text-align: start;
  cursor: default;
  text-transform: none;
}

@media (max-width: 768px) {
  .c1 {
    display: none;
  }
}

<div>
  <span
    alturalinha="24"
    class="c0 c1"
    cor="#793F98"
    cursor="default"
    estilo="semibold"
    tamanho="16"
  />
</div>
`;

exports[`Cabecalho.styles TextoProjeto DEVE ser igual ao snapshot 1`] = `
.c0 {
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  font-family: AvertaStd-Semibold;
  color: #5D615F;
  text-align: start;
  cursor: default;
  text-transform: none;
}

.c2 {
  margin-right: 4px;
}

@media (max-width: 768px) {
  .c1 {
    display: none;
  }
}

@media (max-width: 768px) {
  .c2 {
    display: none;
  }
}

<div>
  <span
    alturalinha="24"
    class="c0 c1 c2"
    cor="#5D615F"
    cursor="default"
    estilo="semibold"
    tamanho="16"
  />
</div>
`;
